### ISSUE TITLE
sybase unicode paramter length

### ISSUE THREAD

**Issue by pomesqb** [linq2db/linq2db#3916]
it's seems that the parameter length generated by paramter content. 
@COD_KIND pass 'QQQ', so length is UniVarChar(3)
@COD_DESC pass unicode content '鏈鋸人唷', so length is UniVarChar(4)
But the unicode content is multi bytes for one character, in our encoding language is 2 bytes for 1 character .
'鏈鋸人唷' in our encoding is 8 bytes, it need UniVarChar(8) to save to database.
So there is an error raise.

i try to add column length to the property, but it not work.
any approach to solve this situation? 
```
 [Column(Length = 8)]
 public virtual string COD_DESC { get; set; }
```

the code:
```
D.ConnLinq.GetTable<COD>()
                    .Where(c => c.COD_ID == "V")
                    .Set(c => c.COD_KIND, "QQQ")
                    .Set(c => c.COD_DESC, "鏈鋸人唷")
                    .Update();
```

the class:
```
public class COD
    {
        [PrimaryKey]
        public virtual string COD_KIND { get; set; } 
        [PrimaryKey]
        public virtual string COD_ID { get; set; } 
        public virtual string COD_DESC { get; set; } 
    }
```

```
DECLARE @COD_KIND UniVarChar(3) 
SET     @COD_KIND = 'QQQ'
DECLARE @COD_DESC UniVarChar(4) 
SET     @COD_DESC = '鏈鋸人唷'
DECLARE @COD_ID_1 UniVarChar(1) 
SET     @COD_ID_1 = 'V'

UPDATE
	[COD]
SET
	[c_1].[COD_KIND] = @COD_KIND,
	[c_1].[COD_DESC] = @COD_DESC,
FROM
	[COD] [c_1]
WHERE
	[c_1].[COD_ID] = @COD_ID_1
```

```
Exception message:Error converting characters into server's character set. Some character(s) could not be converted.
```

### Environment details

`Linq To DB` version: 4.4.0
DataProvider: AdoNetCore.AseClient 0.19.2
Database (with version): Adaptive Server Enterprise/16.0 SP03
Operating system: Window10
.NET Version: .Net Standard 2.0

**Comment by MaceWindu** [comment]
Are you sure it is length issue, not server character set issue? We have unicode tests and they work fine but server should have unicode enabled (utf8 in our case).

What `sp_helpsort` returns in third data set for you?
![image](https://user-images.githubusercontent.com/1099238/212862690-9796f64d-fd0a-41ff-bdc2-ad9fe9560da3.png)


You can also try to execute same query with literal instead of parameter to see if it works. From isql or by setting `db.InlineParameters = true;`

**Comment by MaceWindu** [comment]
Also show `select @@unicharsize` results

**Comment by pomesqb** [comment]
sp_helpsort 
![image](https://user-images.githubusercontent.com/62273346/212888111-7377e4f3-afd6-4bb9-89f1-8fdd4e50e507.png)

select @@unicharsize
![image](https://user-images.githubusercontent.com/62273346/212888206-61d76d20-97b5-45b6-9ff9-66530cfee6f6.png)

yes if i use InlineParameters, it works.
```
db.InlineParameters = true;
```
But our database administrator will not change the server to unicode enabled,
because our database is already 20 years old, they are afraid of changing it.

is the InlineParameters the only way to solve this situation?

**Comment by MaceWindu** [comment]
For now, yes, you should use literals.

We need to think how to fix it.

**Comment by sdanyliv** [comment]
Note that, there is`InlineParameters()` LINQ extension method also.

**Comment by pomesqb** [comment]
ok i got it.
Thanks, you are the best team.

**Comment by MaceWindu** [comment]
As fix we shouldn't apply length to `univarchar`/`unichar` parameters (`DataType.NChar`/`DataType.NVarChar` )

**Comment by MaceWindu** [comment]
Already fixed by #4146
